samples/wifi/softap/src/wpas_events.c

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/net/net_event.h>
+#include <ctrl_iface_zephyr.h>
+#include <supp_events.h>
+#include <supp_main.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+LOG_MODULE_REGISTER(wpas_event, CONFIG_LOG_DEFAULT_LEVEL);
+
+/* TODO: Handle other events */
+#define WPA_SUPP_EVENTS (NET_EVENT_WPA_SUPP_READY)
+
+static struct net_mgmt_event_callback net_wpa_supp_cb;
+
+K_SEM_DEFINE(wpa_supp_ready_sem, 0, 1);
+
+static void handle_wpa_supp_ready(struct net_mgmt_event_callback *cb)
+{
+	k_sem_give(&wpa_supp_ready_sem);
+}
+
+static void wpa_supp_event_handler(struct net_mgmt_event_callback *cb,
+	uint32_t mgmt_event, struct net_if *iface)
+{
+	/* TODO: Handle other events */
+	switch (mgmt_event) {
+	case NET_EVENT_WPA_SUPP_READY:
+		handle_wpa_supp_ready(cb);
+		break;
+	default:
+		LOG_DBG("Unhandled event (%d)", mgmt_event);
+		break;
+	}
+}
+
+int wait_for_wpa_s_ready(void)
+{
+	struct wpa_supplicant *wpa_s = z_wpas_get_handle_by_ifname("nordic_wlan0");
```

Comment:
can `nordic_wlan0` be used as a #define and later used everywhere else?

Reply:
Missed this one, good catch. This is old code copy-pasted again, now we take this from the DTS.